### PR TITLE
docs(x-integration): review build search UI docs

### PR DESCRIPTION
**Pull request by mlcorton**
Full content review regarding integration info for X Components. It involves Build your search UI doc.
- renamed and changed location of architecture md file
- reviewed & adapted structure, voice & style, US English
- fixed links

This PR is caused by [EDOC-1439](https://searchbroker.atlassian.net/browse/EDOC-1439)

# Pull request template

## Motivation and context
Review documentation on how to run the X Components/Archetype project and adapt content to the eDocs style for a better doc structure and understanding.

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: [EDOC-1439](https://searchbroker.atlassian.net/browse/EDOC-1439)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [X] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [ ] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [X] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [X] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:17]: We can include here a info Panel with info like this:

If you are loading an archetype script hosted by empathy, X resources are served by a CDN through the following URLs Environment:

- Live: https://x.empathy.co/{INSTANCE}/app.js
- Staging: https://x.staging.empathy.co/{INSTANCE}/app.js

where {INSTANCE} is the name of your store accorded with Empathy.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:70]:
```suggestion
You can change the snippet configuration values once the project is deployed. Use the `/x-archetype/public/snippet-script.js` file to perform hot changes for the snippet parameters. For more information on the supported parameters, check out [Snippet configuration](#snippet-configuration).
```

File: packages/x-components/static-docs/build-search-ui/README.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/README.md:50]: Should this section be merged with the one below at  **Develop via Interface X Archetype**? It might be weird to explain how to integrate the solution before explaining how to develop the solution.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:162]: You removed the Callbacks section, so now is part of `2. Initialize Interface X` inside `Initializing Interface X project on demand`, but this is not correct, because you can set the callbacks also in `Initializing Interface X project automatically``

I liked more the way it was before, but if you want to separate it, we should have a menu like:

```
## Further considerations

### Snippet Configuration
### Setting callbacks up
### Events
### X API public functions
````

Also the paragraph at the beggining of the current Further Considerations with the three bullets is unnecesary because you are about to read each section. I feel there ir more links and info, but less clear now in this part.

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:162]: FURTHER CONSIDERATIONS neither seems the best title here to me.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:220]: Why are we removing this explanation? I know there are things that you are telling ion other side, but not all of them.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:158]: You have removed the Required column in the new table, and it is much less clear.

File: packages/x-components/static-docs/build-search-ui/web-archetype-development-guide.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-development-guide.md:33]: Does **run** sound better than **execute**? I would also clarify that it is the development server the thing you are running.
```suggestion
2. Install the **project dependencies** and run the project development server.
```

File: packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md:11]: You are calling this section `Use and configure` at README.md, but here you are calling it `How To use and configure`here

File: packages/x-components/static-docs/build-search-ui/web-x-components-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-x-components-integration-guide.md:1]: We should change the name of the file to

`web-x-components-development-guide.md`

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-x-components-integration-guide.md:1]: Remember to change links pointing to it.

**Review by herrardo — CHANGES_REQUESTED** [review]

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:154]: What is this empty link?

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:213]: Are callback and consent links working? I can't find their target, but they seem to link to a random place in the document.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]
GJ!. Ping me if you can before merging the PR and I'll format the documents so it follows our style guidelines.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:213]: There was a callbacks section but Luisa got rid of it, but we are going to restore it, righ @mlcorton ? :)

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:158]: I've removed the _Required_ column to both reduce the length of the overall table and enlarge the width of the _Description_ column. Having too long descriptions makes it difficult to read. However, the information on which param is required or not is still available in the table. This information is not missing but met the same style used in the UI reference tables.  
Due to the huge size of descriptions, I've also relocated some additional information, by removing it from the table and placing it in callouts or other related sections. I've not removed nor deleted any important information, just rewritten or relocated it.

File: packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md:11]: They are different. The individual page/file is called _How to use and configure..._ as we agreed. In the readme we are referring to the section within the getting started page, which doesn't need to be called the same. In addition, in this section, you find a link to the how-to with the appropriate title as follows: `Unsure how to start? Check out [How to use and configure Interface X Components in your project](/develop-empathy-platform/build-search-ui/web-how-to-use-x-components-guide.md).`

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:242]:
```suggestion
| `setSnippetConfig` | [snippet configuration params](#snippet-configuration) - Initialization options | Changes initialization options so that all components react to the changes, i.e. changing both search engine and language without reloading the page. |
| `addProductToCart` | none | To be called from PDP (Product Detail Page) when the user clicks on the Add To Cart Button. Calling this function will send tracking of the AddToCart event to the [Empathy Tagging API](https://docs.empathy.co/develop-empathy-platform/capture-interaction-signals/tagging-api-guide.html) for the product which is currently in screen.  |
```

I am adding this other event because yesterday we realized we didn't document it, and this is basic to integrate customers. cc @javieri-empathy

**Review by herrardo — CHANGES_REQUESTED** [review]

File: packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md:56]: That's why I've transformed it from section to callout (dev tip). If you believe we should get rid of it, I'll remove it :)

File: packages/x-components/static-docs/build-search-ui/README.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/README.md:50]: I totally agree that it's confusing to place the integration doc in the first position when there's a previous step. However, Gerardo wanted it that way... in this file, as it's like a getting started doc, we can change the order and talk first about the development and the integration afterwards. What do you think @herrardo?

Additionally, from my non-tech point of view, it's also weird to have two development types (archetype & library) but just one dedicated integration.

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/README.md:50]: @javieri-empathy I understand your concern, but the thing here is that most of the people reading this section I think is going to be customers that didn't develop the Archetype instance(because we did) and they want just to integrate it on their website. That's why i wanted to put that at the beggining, to not make them feel they are losing something previous that they should read...does it make sense?

File: packages/x-components/static-docs/build-search-ui/web-archetype-development-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-development-guide.md:123]: Done!

File: packages/x-components/static-docs/build-search-ui/web-x-components-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-x-components-integration-guide.md:1]: Done!

File: packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-how-to-use-x-components-guide.md:56]: Done!

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:17]: Just curious, has this instance something to do with the instances a partner can configure with the CMC tool from Playboard, @herrardo? See https://docs-dev.empathy.co/explore-empathy-platform/configure-empathy-platform/request-instance.html

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:17]: I think so, you create the instance, or ask to Empathy to create it for you in fact, and then you set here the InstanceId, but i wouldn't distract this developer with things about the playboard, because Playboard is not a thing that this Frontend integrator developer is going to manage IMO

File: packages/x-components/static-docs/build-search-ui/README.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/README.md:50]: You should take into account that the order in which present the pages in the sidebar doesn't need to match with the one presented in the getting started (readme). IMHO, from my own perspective, the structure may work as follows:

**For the Readme/Getting started:**
* Discover the ecosystem
* Develop using Interface X
  * Develop using the Archetype
  * Develop using the library
* Integrate the Archetype
* Use and configure X Components  

**For the sidebar:**
* Develop using the Archetype
* Integrate the Archetype
* Develop using the library
* How to use X Components

Note: remember that we've defined for the future to have other independent pages as Style your UI and Translate the search experience to cover both the design system & localization

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:213]: My fault! Sorry, I misunderstood the info provided! I'll revert the initial structure

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:242]: To follow the same structure and style before, I'll rewrite the description field: 

> Sends tracking of the `AddToCart`event to the Empathy Tagging microservice for the selected /displayed product.

Can't commit your suggestion directly inhere. If you agree with the proposed description, I'll add the new info in a new commit together with the rest of changes.

Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:242]: To follow the same structure and style before, I'll rewrite the description field: 

> Sends tracking of the `AddToCart`event to the Empathy Tagging microservice for the selected /displayed product.

Can't commit your suggestion directly inhere. If you agree with the proposed description, I'll added the new info in a new commit together with the rest of changes.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:220]: I think I've removed nothing, but rewritten or replaced information. Now, callout information is now available as a stand-alone subsection covering the example and the info related to X events as well.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:17]: Included as a note including a call to support in case of help requesting the instance.

File: packages/x-components/static-docs/build-search-ui/README.md
Comment (javieri-empathy) [reply packages/x-components/static-docs/build-search-ui/README.md]: Understood @herrardo! Makes sense. I'm still not sure this is the best solution, but at least I understand that there is a reason to be sorted that way. FMPOV, i don't see this as a blocker. Can we note this and test it with the first readers of the doc? We can always change it in the future.

@mlcorton So if I understood correctly the sidebar is 100% independent from the markdowns?  I think that is powerful for use cases as the one you propose, but dangerous if it can desync.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:194]:
```suggestion
| `isSpa`                   | `boolean`    | Enables single-page application model. You set to `true` when the X&nbsp;Archetype runs on top of an SPA website.   |
```

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (javieri-empathy) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:248]: I would just add that this function might have a parameter: the product id. If it is not provided X Components will try to use the url as a identifier, but it might fail.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:249]:
```suggestion
| `addProductToCart` | `productId` __Optional__ - The product identifier that has been added to the cart  | Sends tracking of the `AddToCart` event to the [Empathy Tagging microservice](https://docs.empathy.co/develop-empathy-platform/capture-interaction-signals/tagging-api-guide.html) for the product displayed on screen. This function is called from  the product detail page (PDP) when the shopper clicks on the add-to-cart button. If the `productId` is not provided, the URL will be used to detect if the user has discovered the product thanks to a search session. |
```

File: packages/x-components/static-docs/build-search-ui/README.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/README.md]: @javieri-empathy The sidebar is generated automatically based on the folders/files included in `src` from `docs-empathy-documentation` repo. But you can configure the contents and order to show in the sidebar if needed. Check out https://searchbroker.atlassian.net/wiki/spaces/ED/pages/3515187219/Configure+the+automatic+sidebar

File: packages/x-components/static-docs/build-search-ui/README.md
Comment (herrardo) [inline packages/x-components/static-docs/build-search-ui/README.md:55]: Should say here `importing` instead of `integrating`. I am suggesting this because we use integrate for the other section, and here we are inside developing...

```suggestion
- importing the individual **Interface&nbsp;X&nbsp;Components** library in your Vue project for a
```

**Review by herrardo — CHANGES_REQUESTED** [review]
I think we are close to the end @mlcorton !! Good job

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:176]: Guess we're having a linking conflict here among repos. For the link to work properly in code needs the ampersand, but it won't work in eDocs. I'll replace the `&` symbol to avoid conflicts.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md]: Just wondering @herrardo @javieri-empathy, if the parameters for `AddProductToCart` and `Search` functions are optional, then do I guess that parameters for other functions (`setSnippetConfig` or `Init`) are required?? If so, should we say which are required rather than the optional ones? I'm just trying to unify styles, in the Config snippet table and the Props/Slots in UI ref, we indicate the required ones only.

Comment (herrardo) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md]: My answer is going to hurt you. I would inform every one of them with _Optional_ or _Required_because when it was a check on a column, it was obvious, but now it is not so.

Anyway, if you inform only about the Required ones is not a stopper for me.

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md]: Fixed. It works in my local and in eDocs

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (mlcorton) [reply packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md]: Now works in my local and in eDocs

File: packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md
Comment (javieri-empathy) [inline packages/x-components/static-docs/build-search-ui/web-archetype-integration-guide.md:193]:
```suggestion
| `isSpa`                   | `boolean`    |_Optional_. Enables single-page application model. You set it to `true` when the X&nbsp;Archetype runs on top of a SPA website.   |
```

**Review by javieri-empathy — APPROVED** [review]

**Comment by herrardo** [comment]
@javieri-empathy @mlcorton I have pushed the files formatted with prettier, have the final look to approve the PR.

**Comment by mlcorton** [comment]
> 

@herrardo good enough for now :) Thanks, guys, for your revision & time! 🥇

**Review by herrardo — APPROVED** [review]
Waiting for @javieri-empathy approve

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**
